### PR TITLE
fix(wallet): Solana 10x dApp Tx Amount

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Fixes Solana 10x dApp Tx Amount

Resolves <https://github.com/brave/brave-browser/issues/24109>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Go to https://solend.fi/dashboard and connect your wallet.
2. Click on `SOL` and enter an amount in the `Popup` modal, then click the `Supply` button.
3. Review the transaction in the `Wallet` panel, the value sending should not be `10x` the amount you inputed.

Before:

https://user-images.githubusercontent.com/40611140/179564576-8c20c4bc-cc0b-4912-a490-15c26146f886.mov

After:

https://user-images.githubusercontent.com/40611140/179564062-aaa7dab7-892e-4e59-8505-930c8ea420df.mov

### CODE REVIEW

**Comment by bbondy** [comment]
Needs uplift to Release I think

File: components/brave_wallet_ui/common/hooks/transaction-parser.ts [hunk 1]
```diff
@@ -340,7 +340,11 @@ export function useTransactionParser (
           }
         }, new Amount(0)) ?? 0
 
-        const transferedValue = lamportsMovedFromInstructions.toNumber() + value
+        const formattedValue = new Amount(value)
+          .divideByDecimals(selectedNetwork.decimals)
+          .toNumber()
+
+        const transferedValue = (lamportsMovedFromInstructions.toNumber() + formattedValue).toString()
```
Comment (onyb) [inline components/brave_wallet_ui/common/hooks/transaction-parser.ts:347]: Can this work?

```suggestion
        const transferedValue = new Amount(value)
          .divideByDecimals(selectedNetwork.decimals)
          .plus(lamportsMovedFromInstructions)
          .format()
```

**Review by onyb — APPROVED** [review]
++

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.43.36 Chromium: 103.0.5060.114 (Official Build) nightly (64-bit) 
-- | --
Revision | `a1c2360c5b02a6d4d6ab33796ad8a268a6128226-refs/branch-heads/5060@{#1124}`
OS | Linux

- Verified steps from PR 
- Verified Dapp tx amount is shown correctly even for decimal values

https://user-images.githubusercontent.com/17010094/179897118-b481fcab-efd9-401d-8a75-071a097c9256.mp4

**Pull request merged**